src/lib/accessibility/FocusZone/FocusTrapZone.tsx

```diff
@@ -0,0 +1,322 @@
+import * as React from 'react'
+import * as ReactDOM from 'react-dom'
+import * as keyboardKey from 'keyboard-key'
+import * as PropTypes from 'prop-types'
+
+import eventStack from '../../eventStack'
+
+import {
+  getNextElement,
+  getFirstTabbable,
+  getLastTabbable,
+  getWindow,
+  focusAsync,
+  HIDDEN_FROM_ACC_TREE,
+} from './focusUtilities'
+
+import { IFocusTrapZone, IFocusTrapZoneProps } from './FocusTrapZone.types'
+import getUnhandledProps from '../../getUnhandledProps'
+import getElementType from '../../getElementType'
+import * as customPropTypes from '../../customPropTypes'
+
+export class FocusTrapZone extends React.Component<IFocusTrapZoneProps, {}>
+  implements IFocusTrapZone {
+  private static _focusStack: FocusTrapZone[] = []
+  private _root: { current: HTMLElement | null } = { current: null }
+  private _previouslyFocusedElementOutsideTrapZone: HTMLElement
+  private _previouslyFocusedElementInTrapZone?: HTMLElement
+  private windowElement: Window | null
+
+  private createRef = elem => (this._root.current = ReactDOM.findDOMNode(elem) as HTMLElement)
+
+  static propTypes = {
+    as: customPropTypes.as,
+    componentRef: PropTypes.object,
+    className: PropTypes.string,
+    elementToFocusOnDismiss: PropTypes.object,
+    ariaLabelledBy: PropTypes.string,
+    isClickableOutsideFocusTrap: PropTypes.bool,
+    ignoreExternalFocusing: PropTypes.bool,
+    forceFocusInsideTrap: PropTypes.bool,
+    firstFocusableSelector: PropTypes.string,
+    disableFirstFocus: PropTypes.bool,
+    focusPreviouslyFocusedInnerElement: PropTypes.bool,
+  }
+
+  static handledProps = [
+    'as',
+    'className',
+    'componentRef',
+    'elementToFocusOnDismiss',
+    'ariaLabelledBy',
+    'isClickableOutsideFocusTrap',
+    'ignoreExternalFocusing',
+    'forceFocusInsideTrap',
+    'firstFocusableSelector',
+    'disableFirstFocus',
+    'focusPreviouslyFocusedInnerElement',
+  ]
+
+  static defaultProps: IFocusTrapZoneProps = {
+    as: 'div',
+    isClickableOutsideFocusTrap: true,
+  }
+
+  public componentDidMount(): void {
+    FocusTrapZone._focusStack.push(this)
+    const { disableFirstFocus = false } = this.props
+
+    this.windowElement = getWindow(this._root.current)
+    this._previouslyFocusedElementOutsideTrapZone = this._getPreviouslyFocusedElementOutsideTrapZone()
+
+    if (
+      !this._root.current.contains(this._previouslyFocusedElementOutsideTrapZone) &&
+      !disableFirstFocus
+    ) {
+      this.focus()
+    }
+
+    this._hideContentFromAccessibilityTree()
+    this._subscribeToEvents()
+  }
+
+  public render(): JSX.Element {
+    const { className, ariaLabelledBy } = this.props
+    const rest = getUnhandledProps({ handledProps: FocusTrapZone.handledProps }, this.props)
```

Comment:
please, consider to reuse approach suggested in `UIComponent` to fetch handle props, to remove them being hardcoded. Also, it would be great to encapsulate this as a lib function

Reply:
ok, will do that